app/src/main/java/com/sopt/carrot/presentation/profile/ProfileActivity.kt

```diff
@@ -0,0 +1,103 @@
+package com.sopt.carrot.presentation.profile
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
+import com.sopt.carrot.R
+import com.sopt.carrot.databinding.ActivityProfileBinding
+
+class ProfileActivity : AppCompatActivity() {
+
+    private val viewModel: ProfileViewModel by viewModels()
+    private lateinit var binding: ActivityProfileBinding
+
+    private val getImage =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val imageUri = result.data?.data
+                binding.ivProfileImgLoad.setImageURI(imageUri)
+            }
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_profile)
+        binding.viewModel = viewModel
+        binding.lifecycleOwner = this // lifecycleOwner 설정
+
+        with(binding) {
+            // 초기값
+            btnProfileWoman.isSelected = true
+            btnProfileWoman.setTextColor(ContextCompat.getColor(applicationContext, R.color.white))
+            btnProfileAgreeAndApply.isEnabled = false
+        }
+
+        // introductionTextCount 업데이트
+        viewModel.introductionTextCount.observe(this, { textCount ->
+            binding.tvProfileIntroductionTextCount.text = textCount
+        })
+
+        // isButtonEnabled 업데이트
+        viewModel.isButtonEnabled.observe(this, { isEnabled ->
+            binding.btnProfileAgreeAndApply.isEnabled = isEnabled
+        })
+
+        setupClickListeners()
+    }
+
+    private fun setupClickListeners() {
+        with(binding) {
+            // ivProfileImgLoad 터치 시 갤러리 열기
+            ivProfileImgLoad.setOnClickListener {
+                openGallery()
+            }
+            btnProfileMan.setOnClickListener {
+                it.isSelected = true
+                btnProfileMan.setTextColor(
+                    ContextCompat.getColor(
+                        applicationContext,
+                        R.color.white
+                    )
+                )
+                btnProfileWoman.setTextColor(
+                    ContextCompat.getColor(
+                        applicationContext,
+                        R.color.color_sub_gray3
+                    )
+                )
+                btnProfileWoman.isSelected = false
+                viewModel?.selectedGender?.value = 0
+            }
+            btnProfileWoman.setOnClickListener {
```

Comment:
위 아래가 (남/여) 에 따라서 노출되어지는 행위의 결과가 다르기 위해 로직 내 target 만 다를뿐 (남일때는 여, 여일때는 남), 동작하는 로직 자체는 중복되는것같은데요, method 로 빼보는 건 어떨까요?
- 중복 코드 감소
- 가독성 증가

```suggestion
            btnProfileWoman.setOnClickListener {
              btnChange(btnProfileWoman, btnProfileMan)
            }
            
            ...
          
          
       private fun btnChange(btnAsIs: View, btnToBe: View)  {   // 타입이 view 인지는 확인해봐야해요~
         // do something..
       }
```

Reply:
아 정말 그렇네요 코드 중복을 최대한 줄여보겠습니다!